lib/rouge/string_scanner.rb

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+require 'strscan'
+
+module Rouge
+  legacy_specified = "#{ENV['ROUGE_LEGACY_ANCHORS']}" != ""
```

Comment:
In the worst case, anyone can still activate the old behaviour by setting `ROUGE_LEGACY_ANCHORS=1`. I used this to test that all the visual samples (except VimL) are highlighted the same.